### PR TITLE
⬆️(project) migrate to django 4.2

### PR DESCRIPTION
**Pull request by lebaudantoine**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -21,7 +21,7 @@ classifiers =
 [options]
 include_package_data = True
 install_requires =
-    Django<4.2
+    Django==4.2
```
Comment (jmaupetit) [inline setup.cfg]: We usually don't pin dependencies for libraries:

```suggestion
    Django>=4.2
```

Comment (lunika) [reply setup.cfg]: And we should still allow django prior to version 4.2 ? Otherwise project like ashley will not be able to install this lib anymore.

Comment (jmaupetit) [reply setup.cfg]: Afterthought: as we only change things in the sandbox, maybe we can remove constraints on the `Django` dependency for the `django-lti-toolbox` package and pin Django release in the sandbox. WDYT?

Comment (lebaudantoine) [reply setup.cfg]: Following our discussion @jmaupetit, I have fixed up my commit, WDYT? I won't forget to update the initial commit body while squashing my commits.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+### Changed
+
+- Upgrade to Django 4.2
```
Comment (jmaupetit) [inline CHANGELOG.md]: As this change relates to the sandbox, I don't it worth to be mentioned in the CHANGELOG.

**Review by jmaupetit — APPROVED** [review]
A small fix and we are good to go!

File: CHANGELOG.md (hunk 2 above)
Comment (lebaudantoine) [reply CHANGELOG.md]: good catch

File: setup.cfg [hunk 3]
```diff
@@ -41,12 +41,13 @@ dev =
...
     pytest-cov==2.8.1
-    pytest-django==3.8.0
-    pytest==5.3.5
+    pytest-django==4.5.2
+    pytest==7.4.0
 ci =
     twine==2.0.0
 sandbox =
-    django-configurations==2.2
+    Django==4.2
```
Comment (lunika) [inline setup.cfg]:
```suggestion
    Django==4.2.4
```

The 4.2 is a little buggy, you should use the last version.

**Pull request merged**